tests/functional/aws-node-sdk/test/object/listParts.js

```diff
@@ -75,3 +75,154 @@ describe('List parts', () => {
         });
     });
 });
+
+/** Tests for special characters in XML **/
+
+/* eslint-disable no-param-reassign */
+function createParts(sigCfg, bucketUtil, s3, key) {
+    let uploadId;
+    return s3.createBucketAsync({ Bucket: bucket })
+    .then(() => s3.createMultipartUploadAsync({
+        Bucket: bucket, Key: key }))
+    .then(res => {
+        uploadId = res.UploadId;
+        return s3.uploadPartAsync({ Bucket: bucket, Key: key,
+          PartNumber: 1, UploadId: uploadId, Body: bodyFirstPart });
+    })
+    .then(() => Promise.resolve(uploadId));
+}
+/* eslint-enable no-param-reassign */
+function deleteParts(s3, bucketUtil, key, uploadId) {
+    process.stdout.write('Emptying bucket');
+
+    return s3.abortMultipartUploadAsync({
+        Bucket: bucket, Key: key, UploadId: uploadId,
+    })
+    .then(() => bucketUtil.empty(bucket))
+    .then(() => {
+        process.stdout.write('Deleting bucket');
+        return bucketUtil.deleteOne(bucket);
+    });
+}
+
+function test(s3, bucket, key, uploadId, cb) {
+    s3.listParts({
+        Bucket: bucket,
+        Key: key,
+        UploadId: uploadId },
+    (err, data) => {
+        checkNoError(err);
+        assert.strictEqual(data.Key, key);
+        cb();
+    });
+}
+
+describe('List parts - object keys with special characters: `&`', () => {
+    withV4(sigCfg => {
+        const bucketUtil = new BucketUtility('default', sigCfg);
+        const s3 = bucketUtil.s3;
+        let uploadId;
+        const key = '&amp';
+
+        beforeEach(() =>
+            createParts(sigCfg, bucketUtil, s3, key)
+            .then(res => {
+                uploadId = res;
+                return Promise.resolve();
+            })
+        );
+
+        afterEach(() => deleteParts(s3, bucketUtil, key, uploadId));
+
+        it('should list parts of an object with & in it\'s key',
```

Comment:
not sure if you care, but the rest of your it statements use the backticks around the symbol and here they are missing

`it\'s` can just be `its`, the apostrophe is only used for contractions ('it is' --> "it's")